packages/core/src/external/aws/lambda-logic/bundle-to-html.ts

```diff
@@ -1267,12 +1267,14 @@ function createObservationSocialHistorySection(observations: Observation[]) {
   const removeDuplicate = uniqWith(observationsSortedByDate, (a, b) => {
     const aDate = dayjs(a.effectiveDateTime).format(ISO_DATE);
     const bDate = dayjs(b.effectiveDateTime).format(ISO_DATE);
-    const aText = a.code?.text;
-    const bText = b.code?.text;
+    const aText = a.code?.text ?? a.code?.coding?.[0]?.display;
+    const bText = b.code?.text ?? b.code?.coding?.[0]?.display;
+    const aValue = renderSocialHistoryValue(a) ?? "";
+    const bValue = renderSocialHistoryValue(b) ?? "";
```

Comment:
I forgot since I raised this on another PR, but every instance of removeDuplicate we have that looks like this:

const removeDuplicate = uniqWith(conditions, (a, b) => {
    const aDate = dayjs(a.onsetDateTime).format(ISO_DATE);
    const bDate = dayjs(b.onsetDateTime).format(ISO_DATE);
    return aDate === bDate && a.code?.text === b.code?.text;
  })
  
  is buggy since it will render true if the dates are the same and the texts are both undefined.

Reply:
it's a good suggestion - but let's just take it section-by-section and highlight this to others

they all need to be tested individually, and don't want to cause regressions at this point